pandas/tests/groupby/test_timegrouper.py

```diff
@@ -11,6 +11,7 @@
 import pandas as pd
 from pandas import DataFrame, Index, MultiIndex, Series, Timestamp, date_range
 from pandas.core.groupby.ops import BinGrouper
+from pandas.core.resample import TimeGrouper
```

Comment:
Shouldn't we remove this entirely from resample as well?

Reply:
Internally, `TimeGrouper` still holds a lot of the core metadata for resampling. The reason why the `TimeGrouper` isn't needed at the toplevel is because of this shortcut in `Grouper`:

```
    def __new__(cls, *args, **kwargs):
        if kwargs.get('freq') is not None:
            from pandas.core.resample import TimeGrouper
            cls = TimeGrouper
        return super().__new__(cls)

```

So AFAICT, the internal `TimeGrouper` is still needed.

Reply:
see my comments below, we don't want to use this in the user facing tests

Reply:
Would there be any objection to replacing TimeGrouper internally as well? Always looking to reduce GroupBy complexity so getting rid of an entire class would be helpful

Reply:
I doubt this would be easy

TimeGrouper does a lot of stuff

Reply:
Figured as such. Would still be nice if not for a full class removal to even remove any now internally unused methods and keep paring down the groupby code. If I see any opportunities I'll push a PR

Reply:
Definitely +1 for having Grouper adopt TimeGrouper code. Should be possible.